config/main.py

```diff
@@ -1357,12 +1357,67 @@ def apply_patch(ctx, patch_file_path, format, dry_run, ignore_non_yang_tables, i
             patch_as_json = json.loads(text)
             patch = jsonpatch.JsonPatch(patch_as_json)
 
+        results = {}
         config_format = ConfigFormat[format.upper()]
-        GenericUpdater().apply_patch(patch, config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+        if multi_asic.is_multi_asic():
+            # Initialize a dictionary to hold changes categorized by ASIC
+            changes_by_asic = {}
 
+            # Function to extract ASIC identifier from the change path
+            def extract_asic_id(path):
+                start = path.find("/") + 1
+                end = path.find("/", start)
+                return path[start:end], path[end:]  # Also return the modified path without ASIC ID
+
+            # Function to apply patch for a single ASIC.
+            def apply_patch_for_asic(asic_changes):
+                asic_id, changes = asic_changes
+
+                # Replace localhost to empty string which is db definition of Host
+                if asic_id.lower() == "localhost":
+                    asic_id = ""
+
+                try:
+                    # Call apply_patch with the ASIC-specific changes and predefined parameters
+                    GenericUpdater(namespace=asic_id).apply_patch(jsonpatch.JsonPatch(changes), config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+                    results[asic_id] = {"success": True, "message": "Success"}
+
+                    log.log_notice(f"'apply-patch' executed successfully for {asic_id} by {changes}")
+                except Exception as e:
+                    results[asic_id] = {"success": False, "message": str(e)}
+                    log.log_notice(f"'apply-patch' executed failed for {asic_id} by {changes} due to {str(e)}")
+
+            # Iterate over each change in the JSON Patch
+            for change in patch:
+                asic_id, modified_path = extract_asic_id(change["path"])
+
+                # Modify the 'path' in the change to remove the ASIC ID
```

Comment:
>ASIC ID [](http://example.com/codeflow?start=64&length=7)

Regarding the terminology, we can generalize it to "scope" instead of ASIC_ID.
1. whether it is an ASIC or DPU does not matter
2. whether it is ID or name does not matter.

Reply:
This suggestion is only this function.